### PR TITLE
Merge master to refactor

### PR DESCRIPTION
**Pull request by trvrnrth**
Not sure how you'd prefer to handle these merges.

I imagine there are likely to be more changes in this area but getting the two branches in sync first seemed wise.

### CODE REVIEW

File: custom_components/ramses_cc/services.yaml [hunk 1]
```diff
@@ -302,14 +298,19 @@ set_zone_mode:
           mode: slider
 
     duration:
+      name: Duration
```
Comment (zxdavb) [inline custom_components/ramses_cc/services.yaml:301]: Shall it be `Duration (hours/minutes)` too?

File: custom_components/ramses_cc/services.yaml [hunk 2]
```diff
@@ -443,23 +444,23 @@ set_dhw_mode:
         boolean:
 
     duration:
-      name: Override duration
+      name: Duration
```
Comment (zxdavb) [inline custom_components/ramses_cc/services.yaml:447]: See above.

File: custom_components/ramses_cc/services.yaml [hunk 3]
```diff
@@ -443,23 +444,23 @@ set_dhw_mode:
...
-          min: 15
-          max: 1440
-          step: 15
-          unit_of_measurement: mins
-          mode: slider
+        duration:
 
     until:
-      name: Override end time
+      name: Until
```
Comment (zxdavb) [inline custom_components/ramses_cc/services.yaml:458]: `Until (datetime)` ?

**Comment by zxdavb** [comment]
> Not sure how you'd prefer to handle these merges.

We can be adventurous with **refactor**, but keeping sync'd is desirable.

**Comment by zxdavb** [comment]
I am aware my review is not internally consistent,

**Review by zxdavb — APPROVED** [review]

File: custom_components/ramses_cc/services.yaml (hunk 1 above)
Comment (trvrnrth) [reply custom_components/ramses_cc/services.yaml:301]: I didn't change this one before as there isn't the peculiarity of having two separate fields that do more or less the same thing for this service.

Comment (trvrnrth) [reply custom_components/ramses_cc/services.yaml:301]: From what you were saying is it also the case that the underlying will actually respect durations to the second as well? I haven't checked.

File: custom_components/ramses_cc/services.yaml (hunk 3 above)
Comment (trvrnrth) [reply custom_components/ramses_cc/services.yaml:458]: I was figuring that the selector and example were already doing do a good enough job of making it clear.

**Pull request merged**